### PR TITLE
ppc64le support for Quay Operator (PROJQUAY-5372)

### PR DESCRIPTION
**Pull request by jcho02**
Builds and deploys successfully on ppc64le OCP clusters by running `./hack/build.sh` and then `./hack/deploy.sh`
e2e tests pass for all quay-registry builds

Dockerfile supports amd64, ppc64le image operator builds
Makefile runs e2e tests for amd64, ppc64le environments
build.sh has updated amd64, ppc64le supported tag dependencies (quay, quay-builder, quay-builder-qemu clair, redis, postgres)

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -65,13 +65,13 @@ digest "${REGISTRY}/${NAMESPACE}/quay-operator:${TAG}" OPERATOR_DIGEST
 digest "${REGISTRY}/${NAMESPACE}/quay:${TAG}" QUAY_DIGEST
 digest "${REGISTRY}/${NAMESPACE}/clair:nightly" CLAIR_DIGEST
 digest "${REGISTRY}/${NAMESPACE}/quay-builder:${TAG}" BUILDER_DIGEST
-digest "${REGISTRY}/${NAMESPACE}/quay-builder-qemu:main" BUILDER_QEMU_DIGEST
+digest "${REGISTRY}/${NAMESPACE}/quay-builder-qemu:3.9.0" BUILDER_QEMU_DIGEST
 # shellcheck disable=SC2034
-POSTGRES_DIGEST='centos/postgresql-13-centos7@sha256:71b24684d64da46f960682cc4216222a7e4ed8b1a31dd5a865b3e71afdea20d2'
+POSTGRES_DIGEST='quay.io/sclorg/postgresql-15-c9s@sha256:593910f2d4b895f4924261a3b8b2aa6457892100a01a0c0ad661cd378d810d65'
```
Comment (dmage) [inline hack/build.sh]: Quay 3.9 users will use Postgres 13, we cannot switch to Postgres 15 unless we provide automated migration of the data format from 13 to 15.

Comment (jcho02) [reply hack/build.sh]: Changed `POSTRGES_DIGEST` to postgres 13. Left `POSTGRES_UPGRADE_DIGEST`to what it was previously. (doesn't seem to be used by p).

**Comment by jcho02** [comment]
Updated dockerfile, build.sh and image.yaml to support multi-arch builds.
ci.yaml edited build phase for ppc64le builds, removed test for ppc64le as it needs a cluster, tests run successfully locally with correct dependencies.
Removed `s390x` tag to let Z test and add themselves

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi8/go-toolset:1.19 as builder
```
Comment (dmage) [inline Dockerfile:1]: BUILDPLATFROM is the host system platform, i.e. usually it's linux/amd64.

The current version of Dockerfile will create binaries for the host system, not the target system. Either you need to remove `--platform=$BUILDPLATFORM` and rely on emulation, or use TARGETARCH and TARGETOS and rely Go ability to cross-compile.

```
FROM --platform=$BUILDPLATFORM registry.access.redhat.com/ubi8/go-toolset:1.19 AS build
...
ARG TARGETOS TARGETARCH
RUN CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH go build -mod vendor -o manager main.go
```

See https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/ for details.

File: hack/build.sh [hunk 3]
```diff
@@ -58,20 +58,19 @@ function digest() {
...
 digest "${REGISTRY}/${NAMESPACE}/quay-operator:${TAG}" OPERATOR_DIGEST
 
 digest "${REGISTRY}/${NAMESPACE}/quay:${TAG}" QUAY_DIGEST
 digest "${REGISTRY}/${NAMESPACE}/clair:nightly" CLAIR_DIGEST
 digest "${REGISTRY}/${NAMESPACE}/quay-builder:${TAG}" BUILDER_DIGEST
-digest "${REGISTRY}/${NAMESPACE}/quay-builder-qemu:main" BUILDER_QEMU_DIGEST
+digest "${REGISTRY}/${NAMESPACE}/quay-builder-qemu:3.9.0" BUILDER_QEMU_DIGEST
 # shellcheck disable=SC2034
-POSTGRES_DIGEST='centos/postgresql-13-centos7@sha256:71b24684d64da46f960682cc4216222a7e4ed8b1a31dd5a865b3e71afdea20d2'
+POSTGRES_DIGEST='quay.io/sclorg/postgresql-13-c9s@sha256:593910f2d4b895f4924261a3b8b2aa6457892100a01a0c0ad661cd378d810d65'
```
Comment (dmage) [inline hack/build.sh:69]: @jonathankingfc will upgrades work with this image? won't it break downstream?

File: Dockerfile (hunk 2 above)
Comment (jcho02) [reply Dockerfile:1]: Updated dockerfile to have `TARGETOS` and `TARGETARCH` as args

File: hack/build.sh (hunk 3 above)
Comment (jonathankingfc) [reply hack/build.sh:69]: the downstream images are defined in the midstream cpaas repo, so this won't affect the downstream builds

**Review by jonathankingfc — DISMISSED** [review]
LGTM

**Comment by jcho02** [comment]
Finalized build.sh to have proper multi-arch build for operator-index image. 
A single-arch operator-index image will be built bundled by the correct operator-bundle image before manifesting into one multi-arch operator-index image.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -30,16 +33,20 @@ jobs:
...
+      matrix:
+        go-version: ['1.20']
+        platform: ['amd64','ppc64le']
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: OS Dependencies
-        run: apt-get update && apt-get install -y tar make gcc
+        run: apt-get update && apt-get install -y tar make gcc docker.io
```
Comment (yussufsh) [inline .github/workflows/ci.yaml:49]: Is `docker.io` needed now?

File: Dockerfile (hunk 2 above)
Comment (yussufsh) [reply Dockerfile:1]: @jcho02 need to remove `--platform=$BUILDPLATFORM`?

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (dmage) [reply .github/workflows/ci.yaml:49]: we can remove it in a follow-up PR

File: Dockerfile (hunk 2 above)
Comment (jcho02) [reply Dockerfile:1]: no

Comment (jcho02) [reply Dockerfile:1]: no

**Pull request merged**